app/assets/javascripts/govuk_publishing_components/components/layout-super-navigation-header.js

```diff
@@ -4,68 +4,210 @@ window.GOVUK = window.GOVUK || {}
 window.GOVUK.Modules = window.GOVUK.Modules || {};
 
 (function (Modules) {
-  function SuperNavigationToggle ($module) {
-    this.$module = $module
+  var SETTINGS = {
+    breakpoint: {
+      desktop: 768
+    }
+  }
 
-    this.showMenuText = $module.getAttribute('data-text-for-show-menu')
-    this.hideMenuText = $module.getAttribute('data-text-for-hide-menu')
-    this.buttonText = $module.getAttribute('data-text-for-button')
+  // Small helpers that update the label when the state of the button has
+  // changed:
+  var setLabelToHide = function ($button) {
+    var newLabel = $button.getAttribute('data-text-for-hide')
+    $button.setAttribute('aria-label', newLabel)
+  }
+
+  var setLabelToShow = function ($button) {
+    var newLabel = $button.getAttribute('data-text-for-show')
+    $button.setAttribute('aria-label', newLabel)
+  }
 
-    this.$button = this.setupButton($module)
+  // Wrapper functions to contain all of the mechanisms needed for hiding and
+  // toggling the menus.
+  var hide = function ($button, $menu) {
+    $button.setAttribute('aria-expanded', false)
+    $menu.setAttribute('hidden', 'hidden')
```

Comment:
Reading this I can understand a bit more why you went with that `aria-expanded` styling. This is very clean.

Reply:
Having read your commit message I realise I misunderstood what we were talking about the other day and was missing the fact that `hidden` was at play as well.

Reply:
💖